components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigProviderFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.config;
+
+import org.apache.commons.io.FilenameUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.config.provider.ConfigProvider;
+import org.wso2.carbon.config.provider.ConfigProviderImpl;
+import org.wso2.carbon.config.reader.ConfigFileReader;
+import org.wso2.carbon.config.reader.XMLBasedConfigFileReader;
+import org.wso2.carbon.config.reader.YAMLBasedConfigFileReader;
+import org.wso2.carbon.secvault.SecureVault;
+import org.wso2.carbon.secvault.SecureVaultFactory;
+import org.wso2.carbon.secvault.exception.SecureVaultException;
+
+import java.nio.file.Path;
+
+
+/**
+ * This class contains the functionality in managing configuration.
```

Comment:
Since this is a factory class, then the description should reflect that in the java doc comment.